tools/lib/language_indexgen.js

```diff
@@ -0,0 +1,48 @@
+const { basename } = require('path');
+const glob = require('glob');
+
+// if a language name starts with a number, we convert it to a textual representation
+// as variables in JS cannot start with numbers:
+const numbers = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
```

Comment:
I don't love this complexity but see why you feel it's needed since `L_langname` or some such is a bit ugly for external consumption... is there a reason we couldn't just export an object literal with named keys?  I know `rollup` can do dead-code elimination of such things fairly simple.

```
import languages from 'highlight.js/lib/languages';
hljs.registerLanguage("xml", languages.xml)
hljs.registerLanguage("basic", languages.basic)
```

Reply:
Though you're still left with `languages["1c"]` for the outliers I guess vs `.` notation... so both of these are annoying in that we now have document this difference in behavior somewhere... and it's still going to trip someone up who's not expecting it.

Reply:
@joshgoebel actually I wanted to use `_` but changed it due to existing project linting rules turning the whole editor red.

> L_langname or some such is a bit ugly for external consumption.

That would definitely solve the issue at hand (=having aligned, logically named CJS/EJS named exports) in a generic way, albeit an ugly one..

> is there a reason we couldn't just export an object literal with named keys?

For CJS that would work. But `import { 1c } from 'highlight.js/lib/languages'` would throw. And I think `import oneC from './1c'; export default { ['1c']: oneC }` will be handled differently from `export { default as oneC }`  (as in this will probably include the whole default export for EJS builds)